hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestSparkRDDWriteClient.java

```diff
@@ -0,0 +1,124 @@
+
+  static Stream<Arguments> testWriteClientReleaseResourcesShouldOnlyUnpersistRelevantRdds() {
+    return Stream.of(
+        Arguments.of(HoodieTableType.COPY_ON_WRITE, true),
+        Arguments.of(HoodieTableType.MERGE_ON_READ, true),
+        Arguments.of(HoodieTableType.COPY_ON_WRITE, false),
+        Arguments.of(HoodieTableType.MERGE_ON_READ, false)
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource
+  void testWriteClientReleaseResourcesShouldOnlyUnpersistRelevantRdds(HoodieTableType tableType, boolean shouldReleaseResource) throws IOException {
+    final HoodieTableMetaClient metaClient = getHoodieMetaClient(hadoopConf(), URI.create(basePath()).getPath(), tableType, new Properties());
+    final HoodieWriteConfig writeConfig = getConfigBuilder(true)
+        .withPath(metaClient.getBasePathV2().toString())
+        .withAutoCommit(false)
+        .withReleaseResourceEnabled(shouldReleaseResource)
+        .withMetadataConfig(HoodieMetadataConfig.newBuilder().enable(false).build())
+        .build();
+    HoodieTestDataGenerator dataGen = new HoodieTestDataGenerator(0xDEED);
+
+    String instant0 = getCommitTimeAtUTC(0);
+    List<GenericRecord> extraRecords0 = dataGen.generateGenericRecords(10);
+    HoodieJavaRDD<GenericRecord> persistedRdd0 = HoodieJavaRDD.of(jsc().parallelize(extraRecords0, 2));
+    persistedRdd0.persist("MEMORY_AND_DISK", context(), HoodieDataCacheKey.of(writeConfig.getBasePath(), instant0));
+
+    String instant1 = getCommitTimeAtUTC(1);
+    List<GenericRecord> extraRecords1 = dataGen.generateGenericRecords(10);
+    HoodieJavaRDD<GenericRecord> persistedRdd1 = HoodieJavaRDD.of(jsc().parallelize(extraRecords1, 2));
+    persistedRdd1.persist("MEMORY_AND_DISK", context(), HoodieDataCacheKey.of(writeConfig.getBasePath(), instant1));
+
+    SparkRDDWriteClient writeClient = getHoodieWriteClient(writeConfig);
+    List<HoodieRecord> records = dataGen.generateInserts(instant1, 10);
+    JavaRDD<HoodieRecord> writeRecords = jsc().parallelize(records, 2);
+    writeClient.startCommitWithTime(instant1);
+    List<WriteStatus> writeStatuses = writeClient.insert(writeRecords, instant1).collect();
+    assertNoWriteErrors(writeStatuses);
+    writeClient.commitStats(instant1, writeStatuses.stream().map(WriteStatus::getStat).collect(Collectors.toList()),
+        Option.empty(), metaClient.getCommitActionType());
+    writeClient.close();
+
+    if (shouldReleaseResource) {
+      assertEquals(Collections.singletonList(persistedRdd0.getId()),
+          context().getCachedDataIds(HoodieDataCacheKey.of(writeConfig.getBasePath(), instant0)),
+          "RDDs cached for " + instant0 + " should be retained.");
+      assertEquals(Collections.emptyList(),
+          context().getCachedDataIds(HoodieDataCacheKey.of(writeConfig.getBasePath(), instant1)),
+          "RDDs cached for " + instant1 + " should be cleared.");
+      assertTrue(jsc().getPersistentRDDs().containsKey(persistedRdd0.getId()),
```

Comment:
Should we combine these 3 assertions into 1 that asserts all RDDs persisted ids?